src/Xamarin.Android.Build.Tasks/Tasks/AppendCustomMetadataToItemGroup.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Xamarin.Android.Tasks {
+	public class AppendCustomMetadataToItemGroup : Task {
+		[Required]
+		public ITaskItem[] Inputs { get; set; }
+
+		[Required]
+		public ITaskItem[] MetaDataItems { get; set; }
+
+		[Output]
+		public ITaskItem[] Output { get; set; }
+
+		public override bool Execute ()
+		{
+			var output = new List<ITaskItem> ();
+			var metaData = new Dictionary<string, List<ITaskItem>> (StringComparer.InvariantCultureIgnoreCase);
+			foreach (ITaskItem item in MetaDataItems) {
+				if (!metaData.ContainsKey (item.ItemSpec))
+					metaData.Add (item.ItemSpec, new List<ITaskItem> ());
+				metaData[item.ItemSpec].Add (item);
```

Comment:
To reduce operations, we could use TryGetValue.

```csharp
if (!metaData.TryGetValue(item.ItemSpec, out var itemsList)) {
    itemsList = new List<>();
    metaData.Add(itemspec, itemsList);
}
itemsList.Add(item);
```

Reply:
I always forget about `TryGetValue` ... doh